plugins/modules/win_updates.ps1

```diff
@@ -1909,17 +1909,57 @@ namespace Ansible.Windows.WinUpdates
     $installResult = Invoke-AsyncMethod 'Installing updates' $api.InstallAsync($installer, ${function:Receive-CallbackProgress}, $CancelToken)
     $exitResult.changed = $true
 
+    # https://www.microsoft.com/en-us/wdsi/defenderupdates
+    $defenderExe = [System.IO.Path]::Combine($env:ProgramFiles, 'Windows Defender', 'MpCmdRun.exe')
```

Comment:
Would it make more sense to move all this stuff down inside the conditional block where the work is actually done (including deferring the Defender presence detection), so it's all self-contained and only being done when the update in question was included and the other conditions are met? Feels wrong to have it always running.

Reply:
It just means I need to redefine the exe in every loop, this was it's done once and the helper code below is just a scriptblock that is only run in the scenario in question.

Reply:
I means having to do an if check to see if it failed then another if check to see if the exe exists like

```powershell
if ($failedLogic) {
    $defenderExe = [System.IO.Path]::Combine(...)
    if (Test-Path $defenderExe) {
        do defender stuff
    }
}
```

I don't hate it but it's even more indentation inside an already indented section.

Reply:
No worries- might want to revisit abstracting a more generic "recovery tweaks for specific known problems" mechanism if there end up being more of these kinds of things (:crossed_fingers: for not), but totally fine as-is.